### ISSUE TITLE
Aliases to generic type can be created with a `TypeVar` that does not match the generic's declaration

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#4699]
**Describe the bug**
As discovered by @FynnBe, no errors are reported when a type alias for a `Gneric` type is created using a `TypeVar` with the wrong variance.

**To Reproduce**
```python
# pyright: strict
from typing import Generic, TypeVar
from typing_extensions import TypeAlias


Tinv = TypeVar("Tinv", covariant=False, contravariant=False)
Tco = TypeVar("Tco", covariant=True)

class MyBox(Generic[Tinv]):
    pass

CovariantBox: TypeAlias = MyBox[Tco] # no errors, using covariant instead of invariant TypeVar
```

**Expected behavior**
An error message on the last line mentioning  the mismatching variance of the `TypeVar`

**VS Code extension or command-line**
command-line 1.1.295

**Additional context**
`CovariantBox` isn't really seen as covariant on the rest of the code, but it is confusing that this syntax is allowed.

**Comment by erictraut** [comment]
Variance applies only to type variables that are associated with a class, not to type variables associated with type aliases or functions, so I don't consider the code above to be a bug. Mypy doesn't generate an error here either.

However, there is a related bug when a generic type alias is used in a class declaration. The following should generate an error but does not:

```python
# pyright: strict
from typing import Generic, TypeVar, TypeAlias

T = TypeVar("T")
Tco = TypeVar("Tco", covariant=True)

class MyBox(Generic[T]):
    pass

CovariantBox: TypeAlias = MyBox[Tco]

class MyBox2(CovariantBox[Tco]): # This should generate an error
    ...

class MyBox3(MyBox[Tco]): # This correctly generates an error
    ...
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
